app/policies/application_policy.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Base policy class
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
```

Comment:
I think that the most common permissions are (and thus should be the defaults in this parent class):
```

  def index?
    admin_or_owner?
  end

  def create?
    true
  end

  def update?
    admin_or_owner?
  end

  def destroy?
    admin_or_owner?
  end
```

Then the `UserPolicy` only needs to override these:
```
class UserPolicy
  ...
  def index?
    admin?
  end


  # The user is looking at their own record; they 'own' their own record
  def owner?
    user == record
  end

```

Then `BusinessPolicy`  and most (if not all) other Policy classes only need to define their Scope `resolve` method